### PR TITLE
Stable progress

### PR DESCRIPTION
**Pull request by christopherlam**
A few small steps forward. Any comments about the wording of error message?

### CODE REVIEW

**Comment by jralls** [comment]
There's a big difference between unsetting `defer_bal_computation`  and `xaccAccountCommitEdit`: The latter sorts the split list, computes the balance, *and writes out the result to the database*. Since `gnc_gen_trans_list_delete` is a cleanup function called in among other places `on_matcher_cancel_clicked` and `on_matcher_delete_event` that doesn't seem like the right thing to do.

File: gnucash/gnome/gnc-plugin-business.c [hunk 1]
```diff
@@ -715,6 +716,13 @@ gnc_business_assign_payment (GtkWindow *parent,
     if (xaccTransCountSplits(trans) <= 1)
         return;
 
+    if (xaccTransGetFirstAPARAcctSplit (trans, FALSE) == NULL)
+    {
+        gnc_warning_dialog (parent, "%s",
+                            _("There must be an A/P or A/R transfer account."));
+        return;
+    }
+
```
Comment (gjanssens) [inline gnucash/gnome/gnc-plugin-business.c:725]: I don't think this was ever my intention when reworking the Assign As Payment functionality (and this is even against prior advice -- avoid manually creating transactions involving the A/R or A/P accounts).

Why would you prevent such transactions from being considered for being transformed to a business payment ? From my point of view this is safe. If there's no A/R or A/P split, the user can select this in the dialog that will be shown. In that dialog it's the Post to account.

Comment (christopherlam) [reply gnucash/gnome/gnc-plugin-business.c:725]: From my testing it seemed that if there's no APAR split then the assign as payment dialog appears but clicking OK then nothing was actioned in the book.

Comment (gjanssens) [reply gnucash/gnome/gnc-plugin-business.c:725]: That's how it behaves now indeed and that's the bug IMO :)

I'll be looking into this before the 5.1 release. In my first experiment I see there's a currency conversion error showing up in the logs that shouldn't be there. That's my starting point.

Comment (christopherlam) [reply gnucash/gnome/gnc-plugin-business.c:725]: Thanks, let's consider this branch a lost cause then 🙂

**Pull request closed without merging**